### PR TITLE
Feature diff

### PR DESCRIPTION
**Pull request by huynguy97**
Works like a train. Tested big (4k lines) files as well. Also has input validation for security reasons.

### CODE REVIEW

**Comment by jwijenbergh** [comment]
We now really need the totalTabs variable to always increase, see `3f6e157`

**Comment by huynguy97** [comment]
Currently theres a bug when you add tabs, remove tabs and then try to diff. This is because our tab implementation.

**Comment by huynguy97** [comment]
Fixed it, code is still a bit dirty but will clean it up.

File: editor/app.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
-from flask import Flask, render_template, request, redirect
+from flask import Flask, render_template, request, redirect, jsonify
```
Comment (bornobob) [inline editor/app.py:1]: remove jsonify

File: editor/app.py [hunk 2]
```diff
@@ -1,7 +1,8 @@
...
 from flask_bootstrap import Bootstrap
 import os
 import aop
+import json
```
Comment (bornobob) [inline editor/app.py:5]: remove json

File: editor/app.py [hunk 3]
```diff
@@ -29,6 +30,7 @@ def editor():
                                menu_item_html=feature_menu_item_htmls([]),
                                modal_html=feature_modal_htmls([]),
                                **feature_states({}))
+
```
Comment (bornobob) [inline editor/app.py:34]: remove extra new line

File: editor/diff.py [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import difflib
+from flask import jsonify
+
+
```
Comment (bornobob) [inline editor/diff.py:4]: one too many newline

**Review by bornobob — CHANGES_REQUESTED** [review]
Some quite small changes

File: editor/app.py (hunk 1 above)
Comment (huynguy97) [reply editor/app.py]: removed

File: editor/app.py (hunk 2 above)
Comment (huynguy97) [reply editor/app.py:5]: removed

File: editor/templates/diff/index.html [hunk 5]
```diff
@@ -0,0 +1,8 @@
+<div class="d-flex justify-content-center mb-3">
+
```
Comment (Justin-Reniers) [inline editor/templates/diff/index.html:2]: Remove newlines

File: editor/app.py (hunk 3 above)
Comment (huynguy97) [reply editor/app.py:33]: removed

File: aop/extensions/diff.py [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+if feature_diff:
+    print("Using feature: Diff")
+
+    @extends('define_endpoints')
+    def diff_endpoints(flaskapp):
+        @flaskapp.route('/editor/diff', methods=['POST'])
+        def diff_files():
+            incoming = request.get_json()
+            html = df.diffOnSentences(incoming["one"], incoming["two"])
+            return html, 200
```
Comment (Justin-Reniers) [inline aop/extensions/diff.py:19]: Maybe do this with `jsonify`, so you'd get something more akin to `return jsonify(success=False, message='msg: {}'.format(message))`

**Review by bornobob — APPROVED** [review]
Looks good!

**Review by Justin-Reniers — APPROVED** [review]
Looks good!

**Review by bornobob — APPROVED** [review]
Still looks good

**Review by jwijenbergh — APPROVED** [review]
Looks good everybody!

**Pull request merged**